src/Service/PsCountryService.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\PrestaShop\Service;
+
+use Country;
+use Country as PsCountry;
+use MyParcelNL\Pdk\Base\Service\CountryCodes;
+use MyParcelNL\Pdk\Facade\Pdk;
+use MyParcelNL\Pdk\Facade\Platform;
+use MyParcelNL\PrestaShop\Contract\PsCountryServiceInterface;
+use MyParcelNL\Sdk\src\Support\Str;
+
+/**
+ * @template T of PsCountry
+ * @extends \MyParcelNL\PrestaShop\Service\PsSpecificObjectModelService<T>
+ */
+final class PsCountryService extends PsSpecificObjectModelService implements PsCountryServiceInterface
+{
+    /**
+     * @var array<string, int|false>
+     */
+    private static array $countryIdIsoCache = [];
+
+    /**
+     * @TODO: Remove this when Carrier Capabilities service is implemented.
+     *
+     * @param  string $carrierName
+     *
+     * @return array
+     */
+    public function getCountriesForCarrier(string $carrierName): array
+    {
+        // Resolve carrier identifier
+        [$resolvedCarrierName] = explode(':', $carrierName);
+
+        $platform            = Platform::getPlatform();
+        $allCarrierCountries = Pdk::get('countriesPerPlatformAndCarrier')[$platform] ?? [];
+        $countriesForCarrier = $allCarrierCountries[$resolvedCarrierName] ?? [];
+
+        if (true === ($countriesForCarrier['fakeDelivery'] ?? null)) {
+            $allCountries = CountryCodes::ALL;
+        } else {
+            $allCountries = array_merge(
+                $countriesForCarrier['deliveryCountries'] ?? [],
+                $countriesForCarrier['pickupCountries'] ?? []
+            );
+
+            // remove duplicates and sort
+            $allCountries = array_unique($allCountries);
+            sort($allCountries);
+        }
+
+        return $allCountries;
+    }
+
+    /**
+     * @param  string $isoCode
+     *
+     * @return ?int
+     */
+    public function getCountryIdByIsoCode(string $isoCode): ?int
+    {
+        // Also cache "false" values to prevent multiple queries for non-existing countries
+        if (null === (self::$countryIdIsoCache[$isoCode] ?? null)) {
+            self::$countryIdIsoCache[$isoCode] = Country::getByIso(Str::upper($isoCode));
```

Comment:
Waarom heb je er voor gekozen om in de getter de country te fetchen en niet al in de constructor bijvoorbeeld?

Reply:
waarom moet dit in de constructor? het is op basis van de meegegeven param

Reply:
Mijn gedachte was dat je in een getter liever geen properties wil veranderen. Maar in dit geval lijkt dit inderdaad logischer.